Python/sysmodule.c

```diff
@@ -206,9 +201,28 @@ sys_audit_tstate(PyThreadState *ts, const char *event,
     /* Early exit when no hooks are registered */
     PyInterpreterState *is = ts->interp;
     if (!should_audit(is)) {
+        // Validate arguments using assertions, only impact peformance
+        // of Python built in debug mode.
+        assert(event != NULL);
+        assert(!argFormat || !strchr(argFormat, 'N'));
```

Comment:
Why would you put these checks inside here? They get compiled out of release builds entirely, so have no impact, but inside the `if` block we get reduced code coverage.

Reply:
If should_audit() is true, the check are done 4 lines below and raises exception if they fail. I'm not sure how what you mean by code coverage here.

The usage of assertion is to avoid any performance overhead in release mode when no audit hook is installed.